### PR TITLE
Change in shooter angles while moving util

### PR DESCRIPTION
**Pull request by Panda8188**
* Calculate change in "azimuth" angle based on tangential velocity
* Calculate change in "elevation" angle based on radial velocity

For shooter
*TODO: Implement util in shooter code given current tangential, radial velocity, distance from speaker, and current azimuth angle (needed unfortunately). simple tan logic.

google doc is being updated as of this post and will have a detailed explanation regarding all variables and related silly logic

The name "delta angle speed calc util" may be silly goofy, but can be understood as such:
Delta = change in
Angle = angle
(based on)
speed = speed/velocity
calc = calculation
util = utility

So, one can see: The change in angle based on speed is calculated in this utility

### CODE REVIEW

File: src/main/java/frc/robot/utils/DeltaAngleSpeedCalcUtil.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      double distFromSpeakerMeters,
+      double currentAzimuthAngleDegrees) {
+
+
+    /** 
+     * :)))))))))))))))))))
+     * https://docs.google.com/document/d/1rIMmzDA1EMH7NQexIMrNYAD-_KFLe9cL4zPLXdl0lek/edit?usp=sharing
+     */
+    double calculatedAzimuthAngleDegrees = Math.atan(tangentialVelocityMPS / SHOOTER_SPEED_MPS);
+    double deltaAzimuthAngleDegrees = currentAzimuthAngleDegrees - calculatedAzimuthAngleDegrees;
```
Comment (balghane) [inline src/main/java/frc/robot/utils/DeltaAngleSpeedCalcUtil.java]: Isn't `calculatedAzimuthAngleDegrees` itself the delta?

File: src/main/java/frc/robot/utils/DeltaAngleSpeedCalcUtil.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package frc.robot.utils;
+
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.util.Units;
+import frc.robot.Constants;
+
+public class DeltaAngleSpeedCalcUtil {
+  private static final double SPEAKER_HEIGHT_INCHES = Units.inchesToMeters(77.9528);
+  private static final double SHOOTER_SPEED_MPS = Constants.PLACEHOLDER_DOUBLE;
```
Comment (balghane) [inline src/main/java/frc/robot/utils/DeltaAngleSpeedCalcUtil.java]: We're going to have to measure this at some point. For now can you set it to like 10 or something?

Also it won't be possible to unit test without setting this to some non-zero value.

Can you make sure your unit test still passes if we change this number? Like maybe make this public and non-final, and set it in your unit test. Or make this variable and the below function non-static, and set this variable in the constructor. You've got different options.

Comment (Panda8188) [reply src/main/java/frc/robot/utils/DeltaAngleSpeedCalcUtil.java]: Alright

File: src/main/java/frc/robot/utils/DeltaAngleSpeedCalcUtil.java (hunk 1 above)
Comment (Panda8188) [reply src/main/java/frc/robot/utils/DeltaAngleSpeedCalcUtil.java]: Actually I think you're right.. good point.

File: src/main/java/frc/robot/utils/DeltaAngleSpeedCalcUtil.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package frc.robot.utils;
+
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.util.Units;
+import frc.robot.Constants;
+
+public class DeltaAngleSpeedCalcUtil {
+  private static final double SPEAKER_HEIGHT_INCHES = Units.inchesToMeters(77.9528);
```
Comment (avakimc) [inline src/main/java/frc/robot/utils/DeltaAngleSpeedCalcUtil.java]: unless I'm misinterpreting, wouldn't inchesToMeters() return a value in meters? so shouldn't SPEAKER_HEIGHT_INCHES actually be SPEAKER_HEIGHT_METERS?

**Review by avakimc — COMMENTED** [review]
small comment. i am not approving because I don't really understand the math

File: src/main/java/frc/robot/utils/DeltaAngleSpeedCalcUtil.java (hunk 3 above)
Comment (Panda8188) [reply src/main/java/frc/robot/utils/DeltaAngleSpeedCalcUtil.java]: Great point thanks 
I will be making the google doc much more easier to read but here's the link https://docs.google.com/document/d/1rIMmzDA1EMH7NQexIMrNYAD-_KFLe9cL4zPLXdl0lek/edit?usp=sharing
its kinda long so sorry about that

the issue is the diagrams kinda take a little to make because i have to replicate it from my own notes and sometimes my notes are illegible..

**Comment by Panda8188** [comment]
ok so everything kinda died here so im going to fix it all by rebasing off of main which is wisco-cleanup.
